boards/beagle/beagleconnect_freedom/revision.cmake

```diff
@@ -0,0 +1,20 @@
+set(BOARD_REVISIONS "C5" "C7")
+
+# If BOARD_REVISION not set, use the default revision
+if(NOT DEFINED BOARD_REVISION)
+  set(BOARD_REVISION ${LIST_BOARD_REVISION_DEFAULT})
+endif()
+
+if(NOT BOARD_REVISION IN_LIST BOARD_REVISIONS)
+  message(FATAL_ERROR "${BOARD_REVISION} is not a valid revision for BeagleConnect Freedom. Accepted revisions: ${BOARD_REVISIONS}")
+endif()
+
+if("${BOARD_QUALIFIERS}" MATCHES "/cc1352p7$" AND "${BOARD_REVISION}" STREQUAL "C5")
+  set(ACTIVE_BOARD_REVISION "C7")
+  message(WARNING "C5 only supports CC1352P. Using C7 instead.")
+endif()
+
+if("${BOARD_QUALIFIERS}" MATCHES "/cc1352p$" AND "${BOARD_REVISION}" STREQUAL "C7")
+  set(ACTIVE_BOARD_REVISION "C5")
+  message(WARNING "C7 only supports CC1352P7. Using C5 instead.")
+endif()
```

Comment:
this should check from the first character until either the end of the line **or** until a `/`, this code prevents users extending a board out of tree, and it should also not be changing the revision, if there is no revision provided then there can be a default set but do not override what the user has put, throw an error

Reply:
It uses the twister files you have, if there is no .yaml for a board target then twister will not try to use it, and you need to have the revisions in the .yaml files for the twister board name. You should probably not set a default revision anyway since it isn't valid for one of the SoCs, people should provide the correct revision

Reply:
Yes, rename and change like so: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/nordic/nrf9160dk/nrf9160dk_nrf9160_0_7_0.yaml